spec/unit/actions/synchronize_user_filters_spec.rb

```diff
@@ -55,18 +55,26 @@
       context 'when mode is sync_domain_client_workspaces' do
         let(:mode) { 'sync_domain_client_workspaces' }
         let(:clients) { [] }
+        let(:organization_not_in_segment) { double('organization') }
         before do
           allow(domain).to receive(:clients).with(:all, nil).and_return([organization])
           allow(domain).to receive(:clients).with(123_456_789).and_return(organization)
           allow(organization).to receive(:project).and_return(project)
           allow(organization).to receive(:client_id).and_return(123_456_789)
           allow(File).to receive(:open).and_return("client_id\n123456789")
           allow(project).to receive(:deleted?).and_return false
+          allow(organization_not_in_segment).to receive(:segment_uri).and_return('another-segment-uri')
         end
         it 'returns results' do
           result = subject.class.call(params)
           expect(result).to eq(results: [[{}]])
         end
+
+        it 'only uses clients from segments in segments filter' do
```

Comment:
Identical code found in synchronize_user_filters_spec.rb:73, synchronize_users_spec.rb:107 (mass = 72)